metaserver/src/main/scala/scala/meta/languageserver/providers/DocumentFormattingProvider.scala

```diff
@@ -2,31 +2,52 @@ package scala.meta.languageserver.providers
 
 import java.nio.file.Files
 import scala.meta.languageserver.Formatter
+import scala.meta.languageserver.Configuration
 import com.typesafe.scalalogging.LazyLogging
 import langserver.messages.DocumentFormattingResult
 import langserver.types.Position
 import langserver.types.Range
 import langserver.types.TextEdit
 import org.langmeta.inputs.Input
 import org.langmeta.io.AbsolutePath
+import monix.reactive.Observable
+import monix.eval.Task
+
+class DocumentFormattingProvider(
+    configuration: Observable[Configuration],
+    cwd: AbsolutePath
+) extends LazyLogging {
+
+  private lazy val scalafmt = Formatter.classloadScalafmt("1.3.0")
+
+  private def formatterFromConfiguration: Task[Formatter] =
+    configuration.lastL.map { conf =>
```

Comment:
I don't think `lastL` works as we expect it to 🤔 It seems to wait until the Observable completes before completing the Task.

```scala
    import monix.execution.Scheduler.Implicits.global
    import scala.concurrent.duration._
    val o = Observable
      .fromIterable(1.to(10))
      .delayOnNext(500.milliseconds)
    println("STARTING")
    val c = global.scheduleAtFixedRate(Duration.Zero, 1.second) {
      println("Tick")
      o.lastF.foreach(i => println(s"LAST: $i"))
    }
    o.dump("").subscribe()
    Thread.sleep(1000 * 10)
    c.cancel()
```

prints out
```
STARTING
Tick
0: -->1
Tick
1: -->2
2: -->3
Tick
3: -->4
4: -->5
Tick
5: -->6
6: -->7
Tick
7: -->8
8: -->9
Tick
9: -->10
10:  completed
LAST: 10
Tick
LAST: 10
Tick
LAST: 10
Tick
LAST: 10
Tick
LAST: 10
LAST: 10
```

I digged a bit into the monix sources and it seems we need to use MulticastStrategy.Behavior and use `.take(1).lastL`

```scala
    import monix.execution.Scheduler.Implicits.global
    import scala.concurrent.duration._
    val (sub, pub2) = Observable.multicast[Int](MulticastStrategy.behavior(1))
    val pub = pub2.delayOnNext(500.milliseconds)

    var i = 0
    global.scheduleAtFixedRate(Duration.Zero, 500.milliseconds) {
      sub.onNext(i)
      i += 1
    }
    val c = global.scheduleAtFixedRate(Duration.Zero, 1.second) {
      println("Tick")
      pub.take(1).lastL.foreach(i => println(s"LAST: $i"))
    }
    pub.dump("").subscribe()
    Thread.sleep(1000 * 10)
    c.cancel()
```

Produces
```
Tick
0: -->0
LAST: 0
Tick
1: -->1
LAST: 1
2: -->2
Tick
3: -->3
LAST: 3
4: -->4
Tick
5: -->5
LAST: 5
6: -->6
Tick
7: -->7
LAST: 7
8: -->8
Tick
9: -->9
LAST: 9
10: -->10
Tick
11: -->11
LAST: 11
12: -->12
Tick
13: -->13
LAST: 13
14: -->14
Tick
15: -->15
LAST: 15
16: -->16
Tick
17: -->17
LAST: 17
18: -->18
19: -->19
```

Behavior takes an initial value which makes the initial `onNext` unnecessary. This reproduces "reactive variables" according to https://monix.io/docs/2x/reactive/observable-comparisons.html

Reply:
Monix has built-ins for watching files in their monix-nio package.  https://github.com/monix/monix-nio/tree/master/src/main/scala/monix/nio/file

Reply:
@olafurpg thanks for digging into it. I've refactored it to use a `Behavior` strategy. I definitely need to study Monix a bit more to fully grasp these variants.